### PR TITLE
Refactor plugins manager

### PR DESCRIPTION
**Pull request by fowczarek**
- Add missing spans in plugins manager. 
- Optimize import inside plugins manager. 
- Drop plugins manager from settings. 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2057 Huge time consumption for get_product_availability](https://app.clickup.com/t/2549495/SALEOR-2057)

File: saleor/plugins/apps.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from django.utils.module_loading import import_string
+
+
+class PluginConfig(AppConfig):
+    name = "saleor.plugins"
+    verbose_name = "Plugins"
+
+    def ready(self):
+        manager_path = settings.PLUGINS_MANAGER
+        import_string(manager_path)
```
Comment (maarcingebala) [inline saleor/plugins/apps.py:12]: There is a Django checker (see `check_plugins` function) that validates plugin paths in settings. Not sure when AppConfig is run, but I'm wondering if this solution is compatible with the checker? The checker should run first.

**Review by korycins — APPROVED** [review]

File: saleor/plugins/apps.py (hunk 1 above)
Comment (fowczarek) [reply saleor/plugins/apps.py:12]: I will move this validation from checks to AppConfig.ready.

Comment (patrys) [reply saleor/plugins/apps.py:12]: IIRC AppConfig.ready is called before the checks. Not sure we need that checker, though, as long as it's clear which module failed to load.

File: saleor/plugins/checks.py [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-from django.conf import settings
-from django.core.checks import Error, register
-from django.utils.module_loading import import_string
-
-if TYPE_CHECKING:
-    from .base_plugin import BasePlugin
-
-
-@register()
-def check_plugins(app_configs, **kwargs):
```
Comment (korycins) [inline saleor/plugins/checks.py:12]: Have you checked the new flow with cases like - missing/incorrect path to plugin, missing manager, incorrect import inside the plugin,

**Review by korycins — APPROVED** [review]

File: saleor/plugins/manager.py [hunk 3]
```diff
@@ -750,12 +754,7 @@ def external_verify(
...
-) -> PluginsManager:
-    if not manager_path:
-        manager_path = settings.PLUGINS_MANAGER
-    if plugins is None:
-        plugins = settings.PLUGINS
-    manager = import_string(manager_path)
-    return manager(plugins)
+def get_plugins_manager() -> PluginsManager:
+    with opentracing.global_tracer().start_active_span("get_plugins_manager"):
+        manager = import_string(settings.PLUGINS_MANAGER)
```
Comment (patrys) [inline saleor/plugins/manager.py:759]: Do we need `settings.PLUGINS_MANAGER`?

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]
All good, let's just fix the changelog.

**Pull request merged**